### PR TITLE
Create separate `RuntimeTestbed` for each metric

### PR DESCRIPTION
**Pull request by Longarithm**
**Idea**
We currently reuse the same `RuntimeTestbed` for computing metrics in `calls_helper`. Presumably it saved some time (actually not a lot), and allowed us to skip initialization of testbed.

But this led to issues with fees computation: https://github.com/near/nearcore/issues/4482#issuecomment-887633866
Explanation: https://near.zulipchat.com/#narrow/stream/295306-dev-contract-runtime/topic/fees.20.26.20state.20size/near/248169740

So we need to create separate testbeds with different `/tmp/data` folders.

**Testing**
Check for discrepancies in resulting `RuntimeConfig`s

### CODE REVIEW

**Comment by Longarithm** [comment]
Difference between configs without safety multiplier before and after this change.

- `~/tmp/runtime_config_swap_fix.json`: config after last fix in params-estimator https://github.com/near/nearcore/pull/4498
- `~/tmp/runtime_config_no_reuse.json`: config after this fix

```
nearcore % sudo python3 ./runtime/runtime-params-estimator/emu-cost/compare_costs.py ~/tmp/runtime_config_swap_fix.json ~/tmp/runtime_config_no_reuse.json --diff
transaction_costs.data_receipt_creation_config.base_cost.send_sir: first=50739408300812 second=36244799687 second/first=0.00
transaction_costs.data_receipt_creation_config.base_cost.send_not_sir: first=50739408300812 second=36244799687 second/first=0.00
transaction_costs.data_receipt_creation_config.base_cost.execution: first=50739408300812 second=36244799687 second/first=0.00
transaction_costs.data_receipt_creation_config.cost_per_byte.send_sir: first=18343997 second=26011453 second/first=1.42
transaction_costs.data_receipt_creation_config.cost_per_byte.send_not_sir: first=18343997 second=26011453 second/first=1.42
transaction_costs.data_receipt_creation_config.cost_per_byte.execution: first=18343997 second=26011453 second/first=1.42
transaction_costs.action_creation_config.function_call_cost.send_sir: first=275464562500 second=191808375000 second/first=0.70
transaction_costs.action_creation_config.function_call_cost.send_not_sir: first=275464562500 second=191808375000 second/first=0.70
transaction_costs.action_creation_config.function_call_cost.execution: first=275464562500 second=191808375000 second/first=0.70
wasm_config.ext_costs.base: first=106958443 second=151794227 second/first=1.42
wasm_config.ext_costs.storage_write_value_byte: first=1648004295 second=13978988 second/first=0.01
wasm_config.ext_costs.storage_write_evicted_byte: first=2722155471 second=13249396 second/first=0.00
wasm_config.ext_costs.storage_read_key_byte: first=4649624122 second=11106883 second/first=0.00
wasm_config.ext_costs.storage_read_value_byte: first=4425184431 second=2625195 second/first=0.00
wasm_config.ext_costs.storage_remove_key_byte: first=5885197962 second=23594426 second/first=0.00
wasm_config.ext_costs.storage_remove_ret_value_byte: first=5434517167 second=3559096 second/first=0.00
wasm_config.ext_costs.storage_has_key_byte: first=4246523521 second=11002765 second/first=0.00
wasm_config.ext_costs.promise_and_base: first=5480298900 second=558956260 second/first=0.10
wasm_config.ext_costs.promise_and_per_promise: first=43458062 second=2098129 second/first=0.05
wasm_config.ext_costs.promise_return: first=4828728252 second=215863152 second/first=0.04
wasm_config.regular_op_cost: first=1609572 second=2290262 second/first=1.42
```

**Comment by matklad** [comment]
>wasm_config.regular_op_cost: first=1609572 second=2290262 second/first=1.42

This seems strange -- regular_op_cost is a very simple cost (wasm execution itself), it shouldn't depend on the state of the testbed. Do you have any hypothesis as to why you get a 1.42 difference in this case?

**Comment by Longarithm** [comment]
>This seems strange -- regular_op_cost is a very simple cost (wasm execution itself), it shouldn't depend on the state of the testbed. Do you have any hypothesis as to why you get a 1.42 difference in this case?

Nope. I actually thought that it's related to io costs which are not finalized yet.
It's also an interesting coincidence that we have 1.42 difference also for `wasm_config.ext_costs.base` and `transaction_costs.data_receipt_creation_config.cost_per_byte`.

Because this test is quite independent, I'll run it separately.

**Comment by Longarithm** [comment]
@matklad when I recomputed only `regular_op_cost`, I got results close to `1734479` for 3 separate runs.
So `2290262` look incorrect.
On the other hand, what do you think about solving this issue separately? My reasoning:
- intuitively, this PR should not introduce additional instablilty (I will re-run the estimation and check this)
- 1.42 is far from the greatest discrepancies we have between current `RuntimeConfig` and the one from master.

File: Cargo.lock [hunk 1]
```diff
@@ -591,11 +591,11 @@ dependencies = [
...
 name = "borsh"
-version = "0.9.0"
+version = "0.9.1"
 source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "4fcabb02816fdadf90866dc9a7824491ccb63d69f55375a266dc03509ac68d36"
+checksum = "18dda7dc709193c0d86a1a51050a926dc3df1cf262ec46a23a25dba421ea1924"
 dependencies = [
- "borsh-derive 0.9.0",
+ "borsh-derive 0.9.1",
  "hashbrown",
```
Comment (matklad) [inline Cargo.lock:599]: lockfile changes seem unrelated. It's better to avoid drive-by dependency updates.

**Review by matklad — COMMENTED** [review]
>It's also an interesting coincidence that we have 1.42 difference 

what's more, we have a bunch of cases where the difference is 0.7, and 1.42 = (0.7)^-1. 

This seems very peculiar....

**Review by matklad — APPROVED** [review]

**Comment by matklad** [comment]
I think it's good to go as is, but it'd be nice:

* to remove lockfile changes (I *think* they are inessential)
* to re-run the estimator to understand why we see a very stable 1.42 diff on some benches.

**Review by ailisp — APPROVED** [review]
